### PR TITLE
introducing CheckoutConfirmingModal

### PR DESCRIPTION
**Pull request by cellog**
# Description

Based on https://www.figma.com/file/EZ4mtBKGj54EQCQvYDArtqKs/Forbes-Crypto-Blockchain?node-id=9%3A16

This new component will be the "optimistic unlocking modal" that makes it far clearer to users that they can start reading stuff after buying a key on the ad remover paywall. This first iteration is limited to that scope. A future PR will make the text customizable so we can use the same view for other apps like regular paywall.

@smombartz I made a basic mobile view, would love some feedback on that too.

Screen Shot 2019-07-18 at 11 32 46 AM
Screen Shot 2019-07-18 at 11 32 44 AM
Screen Shot 2019-07-18 at 11 32 41 AM
Screen Shot 2019-07-18 at 11 20 08 AM
Screen Shot 2019-07-18 at 11 20 06 AM
Screen Shot 2019-07-18 at 11 19 39 AM

# Issues

Refs #4238 

# Checklist:

- [x] 1 PR, 1 purpose: my Pull Request applies to a single purpose
  - [ ] This PR only contains configuration changes (package.json, etc.)
  - [x] This PR only contains code changes (if configuration changes are required, do a separate PR first, then re-base)
- [x] My code follows the style guidelines of this project, including naming conventions
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] If my code adds or changes components, I have written corresponding stories with Storybook
- [x] I have performed a self-review of my own code
- [x] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

**Comment by julien51** [comment]
Great! I was about to suggest something like this because it will make things easier for Forbes!

**Comment by julien51** [comment]
Can you clarify the behavior? If the user clicks on "start reading" what happens? And if they re-open right after?
Also, how does it look with several other locks?

File: paywall/src/components/checkout/CheckoutConfirmingModal.tsx [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+          lock={noTitleLock}
+          account={account}
+          disabled
+          purchase={() => {}}
+          hideCheckout={hideCheckout}
+        />
+        <div>
+          <p>
+            Your transaction was sent and is currently being confirmed on the
+            Ethereum blockchain. Feel free to start browsing our site ads free!
```
Comment (cnasc) [inline paywall/src/components/checkout/CheckoutConfirmingModal.tsx]:
```suggestion
            Ethereum blockchain. Feel free to start browsing our site ad-free!
```

Wording nitpick, but I see "ad-free" in common use as opposed to "ads free".

Comment (julien51) [reply paywall/src/components/checkout/CheckoutConfirmingModal.tsx]: This message is specific to Forbes so it should be part of the `unlockProcolConfig` but we should provide a more "gemeric" default (one which does not mention ad free...)

**Comment by cellog** [comment]
> Can you clarify the behavior? If the user clicks on "start reading" what happens? And if they re-open right after?
> Also, how does it look with several other locks?

Yes.

The flow is:

--> Checkout modal with all locks displayed
--> User clicks a lock to initiate purchase
--> User confirms in their wallet
--> This screen opens while the transaction is submitted/pending/confirming
--> If user watches it all the way to the end, it switches back to the original modal, displaying all the locks again with the expiration date or time

If the user clicks the big green button, the modal disappears, just like it does now with the multi-lock modal

**Comment by julien51** [comment]
Thanks for the clarifications, can you expand on the closing/re-open behavior and the multi locks?

**Comment by julien51** [comment]
Also when you say 
>  This screen opens while the transaction is submitted/pending/confirming
I assume it replaces the regualr checkout modal rather than open on top of it?

**Comment by cellog** [comment]
Purchase flow, visually:

Screen Shot 2019-07-18 at 11 47 17 AM
Screen Shot 2019-07-18 at 11 47 10 AM
Screen Shot 2019-07-18 at 11 47 03 AM
Screen Shot 2019-07-18 at 11 46 47 AM

The user can dismiss any of these modals at any time by clicking the "X" in the upper right corner. On this new Confirming modal screen, they can dismiss it by clicking the big green "Start Reading" button. To re-open the modal, the site has to provide a button to show it again (like the long green button Forbes is using)

**Comment by cellog** [comment]
> Also when you say
> 
> > This screen opens while the transaction is submitted/pending/confirming
> I assume it replaces the regualr checkout modal rather than open on top of it?

Yes, exactly!

**Comment by julien51** [comment]
💯 
> To re-open the modal, the site has to provide a button to show it again (like the long green button Forbes is using)
Do you confirm when they re-open that they do not see the "CheckoutConfirmingModal" ?

**Comment by cellog** [comment]
> 💯
> 
> > To re-open the modal, the site has to provide a button to show it again (like the long green button Forbes is using)
> 
> Do you confirm when they re-open that they do not see the "CheckoutConfirmingModal" again?

The code has no opinion yet, so we could do that. I had imagined that it would show it until the key purchase is confirmed, but if we wish to make it only show once, and then go back to the normal modal, that is easy enough to implement, so sure :)

**Comment by cellog** [comment]
@cnasc @julien51 made that last controversial sentence significantly less controversial, take a look when you get a moment :)

**Comment by julien51** [comment]
@cellog if you're around let's merge.

**Pull request merged**